tools/build/compute_version.go

```diff
@@ -0,0 +1,80 @@
+// Package main is a tool to compute the version of the project.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+func main() {
+	var numeric bool
+	flag.BoolVar(&numeric, "numeric", false, "Print a numeric version")
+	flag.Parse()
+
+	if numeric {
+		fmt.Println(computeNumericVersion())
+		return
+	}
+
+	fmt.Println(computeFullVersion())
+}
+
+func computeFullVersion() string {
+	tag, err := getTag()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	if dirty, err := isDirty(); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	} else if dirty {
+		tag += "-dirty"
+	}
+
+	return tag
+}
+
+func computeNumericVersion() string {
+	tag, err := getTag()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+
+	expr := regexp.MustCompile(`^v(\d+\.\d+\.\d+)-`)
+	matches := expr.FindStringSubmatch(tag)
+	if len(matches) != 2 {
+		fmt.Fprintf(os.Stderr, "Error: tag %s does not match the expected format\n", tag)
+		os.Exit(1)
+	}
+
+	return matches[1]
+}
+
+func getTag() (string, error) {
+	// Note: we cannot use --dirty because it does not detect untracked files.
+	out, err := exec.Command("git", "describe", "--tags", "HEAD").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("git describe: %v. %s", err, out)
+	}
+
```

Comment:
TBH, I would be more leniant and this would be `getTagVersion` and removing any `v` as prefix. Then, you simplify the regexp.

Reply:
Removed the `v` from the version regex 👍